hls4ml/templates/vivado/nnet_utils/nnet_common.h

```diff
@@ -2,6 +2,7 @@
 #define NNET_COMMON_H_
 
 #include "ap_fixed.h"
+#include "nnet_helpers.h"
```

Comment:
`nnet_helpers.h` (mostly) doesn't contain synthesizable code and it is not intended to be included by other files apart from the testbench. And I don't see you using anything from it in the code that is introduced

Reply:
If I remove that, I get this error on my EL 8 machine with g++ 8.5.0
```
In file included from firmware/nnet_utils/nnet_conv1d_latency.h:4,
                 from firmware/nnet_utils/nnet_code_gen.h:4,
                 from firmware/parameters.h:7,
                 from firmware/myproject.cpp:4:
firmware/nnet_utils/nnet_common.h: In function ‘T nnet::reduce(const T*, Op)’:
firmware/nnet_utils/nnet_common.h:37:39: error: there are no arguments to ‘floorlog2’ that depend on a template parameter, so a declaration of ‘floorlog2’ must be available [-fpermissive]
     static constexpr int leftN = pow2(floorlog2(N - 1)) > 0 ? pow2(floorlog2(N - 1)) : 0;
                                       ^~~~~~~~~
firmware/nnet_utils/nnet_common.h:37:39: note: (if you use ‘-fpermissive’, G++ will accept your code, but allowing the use of an undeclared name is deprecated)
firmware/nnet_utils/nnet_common.h:37:68: error: there are no arguments to ‘floorlog2’ that depend on a template parameter, so a declaration of ‘floorlog2’ must be available [-fpermissive]
     static constexpr int leftN = pow2(floorlog2(N - 1)) > 0 ? pow2(floorlog2(N - 1)) : 0;
```